### PR TITLE
Move Scales to redux

### PR DESCRIPTION
**Pull request by deathbearbrown**
This is part one of the refactor.  It took a while because I kept finding bugs with scales in general. 
Some of them are reflected in issue #375 

And additional bug/problem is how we want to handle scales when we've already added x1, x2, y1, y2.  Can you have more scales than that??

To test this, add data to a mark (test for multiple axi) and you should see scales show up in the sidebar.

I haven't added tests yet because I want to refactor this more.

Next I am going to add:
- clearing out/ removing scales
- a simple inspector

### CODE REVIEW

**Comment by arvind** [comment]
> And additional bug/problem is how we want to handle scales when we've already added x1, x2, y1, y2. > Can you have more scales than that??

Yes, you can have an unlimited number of scales (x, y, etc.). E.g., you may have multiple different marks drawing from different datasets as part of the same visualization.

File: src/js/components/visualization/ScaleList.jsx [hunk 1]
```diff
@@ -7,24 +7,30 @@ var React = require('react'),
...
 function mapStateToProps(reduxState, ownProps) {
   return {
-    selectedId: reduxState.get('selectedMark')
+    selectedId: reduxState.get('selectedMark'),
+    scales: reduxState.get('scales')
   };
 }
 
+function mapDispatchToProps(dispatch) {
+  return {};
```
Comment (arvind) [inline src/js/components/visualization/ScaleList.jsx:16]: If this is empty, do we still need it?

**Comment by arvind** [comment]
One functional bug I've noticed: if I drag out a rect mark and then some ordinal data (e.g., gapminder > country) to x for example, the inspector updates to show the scale/field for "left", but "right" doesn't update to "width" and automatic. On lyra2, this works correctly after accounting for the "forceUpdate" (e.g., after closing out the hints box). This might be related to #375 or even #343?

Except for this, 👍 for functionality.

untitled

File: src/js/components/visualization/ScaleList.jsx (hunk 1 above)
Comment (deathbearbrown) [inline src/js/components/visualization/ScaleList.jsx:16]: no, it's just a place holder for when you change the scale name.

**Pull request merged**